Makefile

```diff
@@ -0,0 +1,270 @@
+# found an issue.
+# TODO: do not ignore GHSA-hcpj-qp55-gfph once the patch is out.
+# See: https://github.com/gitpython-developers/GitPython/issues/1515.
+.PHONY: audit
+audit:
+	if ! $$(python -c "import pip_audit" &> /dev/null); then \
+	  echo "No package pip_audit installed, upgrade your environment!" && exit 1; \
+	fi;
+	python -m pip_audit --skip-editable --desc on --fix --dry-run --ignore-vuln GHSA-hcpj-qp55-gfph
+
+# Run some or all checks over the package code base.
+.PHONY: check check-code check-bandit check-flake8 check-lint check-mypy check-go
+check-code: check-bandit check-flake8 check-lint check-mypy check-go
+check-bandit:
+	pre-commit run bandit --all-files
+check-flake8:
+	pre-commit run flake8 --all-files
+check-lint:
+	pre-commit run pylint --all-files
+check-mypy:
+	pre-commit run mypy --all-files
+check-go:
+	pre-commit run golangci-lint --all-files
+	pre-commit run go-build-mod --all-files
+	pre-commit run go-build-repo-mod --all-files
+	pre-commit run go-mod-tidy --all-files
+	pre-commit run go-mod-tidy-repo --all-files
+	pre-commit run go-test-mod --all-files
+	pre-commit run go-test-repo-mod --all-files
+	pre-commit run go-vet-mod --all-files
+	pre-commit run go-vet-repo-mod --all-files
+	pre-commit run go-fmt --all-files
+	pre-commit run go-fmt-repo --all-files
+check:
+	pre-commit run --all-files
+
+
+# Run all unit tests. The --files option avoids stashing but passes files; however,
+# the hook setup itself does not pass files to pytest (see .pre-commit-config.yaml).
+.PHONY: test
+test: test-go
+	pre-commit run pytest --hook-stage push --files tests/
+test-go:
+	go test ./golang/...
+
+# Run the integration tests.
+.PHONY: integration-test
+integration-test:
+	scripts/dev_scripts/integration_tests.sh $(MACARON_PATH) $$HOME
+
+# Build a source distribution package and a binary wheel distribution artifact.
+# When building these artifacts, we need the environment variable SOURCE_DATE_EPOCH
+# set to the build date/epoch. For more details, see: https://flit.pypa.io/en/latest/reproducible.html
+.PHONY: dist
+dist: dist/$(PACKAGE_NAME)-$(PACKAGE_VERSION)-py3-none-any.whl dist/$(PACKAGE_NAME)-$(PACKAGE_VERSION).tar.gz dist/$(PACKAGE_NAME)-$(PACKAGE_VERSION)-docs-html.zip dist/$(PACKAGE_NAME)-$(PACKAGE_VERSION)-build-epoch.txt
+dist/$(PACKAGE_NAME)-$(PACKAGE_VERSION)-py3-none-any.whl: check test integration-test
+	flit build --setup-py --format wheel
+dist/$(PACKAGE_NAME)-$(PACKAGE_VERSION).tar.gz: check test
+	flit build --setup-py --format sdist
+dist/$(PACKAGE_NAME)-$(PACKAGE_VERSION)-docs-html.zip: docs
+	python -m zipfile -c dist/$(PACKAGE_NAME)-$(PACKAGE_VERSION)-docs-html.zip docs/_build/html
+dist/$(PACKAGE_NAME)-$(PACKAGE_VERSION)-build-epoch.txt:
+	echo $(SOURCE_DATE_EPOCH) > dist/$(PACKAGE_NAME)-$(PACKAGE_VERSION)-build-epoch.txt
+
+# Build the HTML documentation from the package's source.
+.PHONY: docs
+docs: docs/_build/html/index.html
+docs/_build/html/index.html: check test
```

Comment:
I don't think we should run the check and test whenever we want to build the documentation as it will take a lot of time. For example, a contributor when wanting to update the RST files in the `docs` directory shouldn't worry about getting the `check` and `test` but only care about whether the building the documentation pass. 

However, always try to build the docs before pushing might be okay as we want to avoid inconsistent in API documentation.

Reply:
> I don't think we should run the check and test whenever we want to build the documentation as it will take a lot of time. For example, a contributor when wanting to update the RST files in the `docs` directory shouldn't worry about getting the `check` and `test` but only care about whether the building the documentation pass.
> 
> However, always try to build the docs before pushing might be okay as we want to avoid inconsistent in API documentation.
Yes, I see your point. But we want to make sure we generate docs when all checks and tests pass. What we can do locally is to comment out the `test` and `check` here if we just want to test creating `docs`.